### ISSUE TITLE
Query about expected iedit-mode-toggle-on-function behaviour

### ISSUE THREAD

**Issue by T-J-Teru** [victorhge/iedit#107]
At one point in the past I'm sure I remember `iedit-mode-toggle-on-function` "toggling" selections within the current function.  So, say I have `test.c` containing:
```
int
foo (arg)
{
  arg + 2;
  return arg;
}
```

And I place the cursor on the first `arg`, then run `iedit-mode-toggle-on-function`, all of the instances of `arg`  within `foo` are highlighted.

Now I seem to remember that at one point running `iedit-mode-toggle-on-function` again would leave iedit mode, unhighlighting all of the `arg`s.  Now what I see is a message: `Restricted in current function, 3 matches.`.

What is the expected behaviour here?  I guess I can achieve the effect of "toggling" by binding `iedit-mode-toggle-on-function` to enter `iedit` mode, then bind a different function to the same key-cord in order to leave iedit mode, but I thought I'd check if the current behaviour is expected or not.

**Comment by victorhge** [comment]
It is not expected, as “toggle" is part of the name.  I will fix it.

**Comment by thierryvolpiatto** [comment]
Here what I am using instead if that helps:
```elisp
(defun iedit-narrow-to-defun (arg)
    (interactive "P")
    (require 'iedit)
    (save-window-excursion
      (save-restriction
        (narrow-to-defun)
        (iedit-mode arg))))
```

**Comment by victorhge** [comment]
Fixed in commit 7eef34898f64ef57078a65737e12f8bd355ec389.
